src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -185,16 +209,34 @@ export class MeetingManager implements AudioVideoObserver {
     configuration: MeetingSessionConfiguration,
     deviceLabels: DeviceLabels | DeviceLabelTrigger = DeviceLabels.AudioAndVideo,
   ): Promise<any> {
-    const { logger: inputLogger, simulcastEnabled, videoDownlinkBandwidthPolicy } = this.config;
-    const logger = inputLogger ? inputLogger : this.createLogger(configuration);
-
+    const {
+      simulcastEnabled,
+      logger: configLogger,
+      logLevel,
+      postLogConfig,
+      videoDownlinkBandwidthPolicy
+    } = this.meetingManagerConfig;
+    
+    // We are assigning the values from the `meetingManagerConfig` to preserve backward compatibility.
+    // Please use `this.meetingManagerConfig` for any values going forward.
     if (simulcastEnabled) {
+      this.simulcastEnabled = simulcastEnabled;
       configuration.enableUnifiedPlanForChromiumBasedBrowsers = true;
       configuration.enableSimulcastForUnifiedPlanChromiumBasedBrowsers = true;
-    };
+    }
+    if (configLogger) {
+      this.logger = configLogger;
+    } else {
+      this.logLevel = logLevel;
+      if (postLogConfig) {
+        this.postLoggerConfig = postLogConfig;
+      }
+    }
```

Comment:
Before we assign the value of `config.xxx` to `this.xxx` in the constructor, which means `this.xxx` get assigned when the App is opened (before calling `join` method). But now they are assigned after `join` is called.

Will this affect backward compatibility? Do we need to keep the original constructor code?

Reply:
Yeah that makes sense actually, builders may be using the variables before they call join(), I do not know what the use-case would be but I will keep the constructor code. Nice call!